addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -1879,6 +1895,64 @@ MockServer.include({
         ]);
         this._mockUnlink(model, [followers.map(follower => follower.id)]);
     },
+    /**
+     * Simulates `create_tracking_values` on `mail.tracking.value`
+     */
+     _mockMailTrackingValue_CreateTrackingValues(values) {
+        switch (values.field_type) {
+            case 'datetime':
+            case 'float':
+            case 'integer':
+            case 'monetary':
+            case 'text':
+                values[`old_value_${values.field_type}`] = values['old_value'];
+                values[`new_value_${values.field_type}`] = values['new_value'];
+                break;
+            case 'date':
+                values[`old_value_${values.field_type}`] = datetime_to_str(values['old_value']);
+                values[`new_value_${values.field_type}`] = datetime_to_str(values['new_value']);
+                break;
+            case 'boolean':
+                values['old_value_integer'] = values['old_value'];
+                values['new_value_integer'] = values['new_value'];
+                break;
+            default:
+                values['old_value_char'] = values['old_value'];
+                values['new_value_char'] = values['new_value'];
+        }
+        delete values['old_value'];
+        delete values['new_value'];
+        return values;
+    },
+    /**
+     * Simulates `get_display_value` on `mail.tracking.value`
+     */
+    _mockMailTrackingValue_GetDisplayValue(record, type) {
+        switch (record.field_type) {
+            case 'float':
+            case 'interger':
+            case 'monetary':
+            case 'text':
+                return record[`${type}_value_${record.field_type}`];
+            case 'datetime':
+                if (record[`${type}_value_datetime`]) {
+                    const datetime = record[`${type}_value_datetime`];
+                    return `${datetime}Z`;
+                } else {
+                    return record[`${type}_value_datetime`];
+                }
+            case 'date':
+                if (record[`${type}_value_datetime`]) {
+                    return date_to_str(record[`${type}_value_datetime`]);
+                } else {
+                    return record[`${type}_value_datetime`];
+                }
+            case 'boolean':
+                return !!record[`${type}_value_integer`];
+            default:
+                return record[`${type}_value_char`];
```

Comment:
should have the same shape as server code, char is in the first condition and there is no default

Reply:
also missing `selection` explicitly somewhere

Reply:
and `many2one`

Reply:
Actually my bad I think I mixed up create and display.

Reply:
Well it looks like a mix of both, not all of them are good but not all of them are wrong, double check it to better match was is done.

Reply:
I don't really agree :) it is a direct adaptation of the `get_display_value` server method.
- There is no default on the server but the last else is kind of a default..
- There is no char in the first condition, but the default already maps the display value to `type_value_char` so it would be redundant (I can add it to be more explicit if you want me to ?)

I feel like the flow is exactly the same :
- text, monetary, float, integer, char 
 ```javascript
// javascript
return record[`${type}_value_${record.field_type}`]
```
```python
# python
getattr(record, '%s_value_%s' % (type, record.field_type)
```
- datetime (date is almost the same)
 ```javascript
// javascript
if (record[`${type}_value_datetime`]) {
     const datetime = record[`${type}_value_datetime`];
     return `${datetime}Z`;
} else {
    return record[`${type}_value_datetime`];
}
```
```python
# python
if record['%s_value_datetime' % type]:
    new_datetime = getattr(record, '%s_value_datetime' % type)
    result.append('%sZ' % new_datetime)
 else:
    result.append(record['%s_value_datetime' % type])
```
- boolean
 ```javascript
// javascript
 return !!record[`${type}_value_integer`];
```
```python
# python
result.append(bool(record['%s_value_integer' % type]))
```
- else
 ```javascript
// javascript
return record[`${type}_value_char`];
```
```python
# python
result.append(record['%s_value_char' % type])
```

Reply:
Yes I know I mixed up both of them. Just make sure to stay as close to python as possible and it should be fine.

Double check especially the create method, which I think is further from the python than the display method on which I posted the comment.